plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-button/payment-gateway-button.tsx

```diff
@@ -30,18 +34,22 @@ export const PaymentGatewayButton = ( {
 	textNeedsSetup?: string;
 } ) => {
 	const { createErrorNotice } = dispatch( 'core/notices' );
-	const { enablePaymentGateway } = useDispatch( PAYMENT_SETTINGS_STORE_NAME );
+	const { enablePaymentGateway, invalidateResolutionForStoreSelector } =
+		useDispatch( PAYMENT_SETTINGS_STORE_NAME );
+	const [ isUpdating, setIsUpdating ] = useState( false );
 
-	const { isUpdating, shouldRedirect } = useSelect( ( select ) => {
-		return {
-			isUpdating: select( PAYMENT_SETTINGS_STORE_NAME ).isGatewayUpdating(
-				id
+	const createAPIErrorNotice = () => {
```

Comment:
Nitpick: I think the convention in WooCommerce Core is to use camelCase even for acronyms, so this should be `createApiErrorNotice`, in this case. 🙂

Reply:
thanks! fixed in https://github.com/woocommerce/woocommerce/pull/52824/commits/08d254936ec3787441686c3a6103e3735f4e6d82